lnd-rpc/src/main/scala/org/bitcoins/lnd/rpc/LndRpcClient.scala

```diff
@@ -0,0 +1,517 @@
+package org.bitcoins.lnd.rpc
+
+import akka.actor.ActorSystem
+import akka.grpc.{GrpcClientSettings, SSLContextUtils}
+import com.google.protobuf.ByteString
+import grizzled.slf4j.Logging
+import lnrpc._
+import org.bitcoins.commons.jsonmodels.lnd._
+import org.bitcoins.core.currency._
+import org.bitcoins.core.number.UInt32
+import org.bitcoins.core.protocol.BitcoinAddress
+import org.bitcoins.core.protocol.ln.LnInvoice
+import org.bitcoins.core.protocol.ln.currency.MilliSatoshis
+import org.bitcoins.core.protocol.ln.node.NodeId
+import org.bitcoins.core.protocol.script.ScriptPubKey
+import org.bitcoins.core.protocol.transaction.TransactionOutPoint
+import org.bitcoins.core.util.StartStopAsync
+import org.bitcoins.core.wallet.fee.SatoshisPerByte
+import org.bitcoins.crypto.DoubleSha256DigestBE
+import org.bitcoins.lnd.rpc.LndRpcClient._
+import org.bitcoins.lnd.rpc.config.LndInstance
+import org.bitcoins.rpc.util.NativeProcessFactory
+import scodec.bits.ByteVector
+import walletrpc.WalletKitClient
+
+import java.io.{File, FileInputStream}
+import java.net.InetSocketAddress
+import java.util.concurrent.atomic.AtomicInteger
+import scala.concurrent.duration.{DurationInt, FiniteDuration}
+import scala.concurrent.{ExecutionContext, Future, Promise}
+import scala.util.{Failure, Success, Try}
+
+/** @param binary Path to lnd executable
+  */
+class LndRpcClient(val instance: LndInstance, binary: Option[File] = None)(
```

Comment:
Back to API discussion, is there some subset of APIs that are common across Eclair and Lnd? 

We could create some sort of `LightningRpcApi` that is extended by both this and `EclairRpcClient`. 

That way, when building lightning applications we can be more abstract rather than coding to a specific implementation. 

Here is `EclairApi` 

https://github.com/bitcoin-s/bitcoin-s/blob/5b762fb2e13742f057162e14a035b67730053ca0/eclair-rpc/src/main/scala/org/bitcoins/eclair/rpc/api/EclairApi.scala#L29

Reply:
This sounds good in theory but don't think will work out to well. These 2 return and take in a lot of different types for matching functions (ie eclair returns a `FundedChannelId` for openChannel and lnd returns a `Option[TransactionOutpoint]`) so I don't know how useful we could actually make it

Reply:
I'll file an issue to think about this further. We can make their API conform to our API in some cases, but the example you cite is a good one, if `None` is given to us we can't translate that into a `FundedChannelId` :/

Reply:
https://github.com/bitcoin-s/bitcoin-s/issues/2844